apps/src/lib/client/tx.rs

```diff
@@ -532,7 +531,59 @@ pub async fn submit_init_proposal(mut ctx: Context, args: args::InitProposal) {
     let proposal: Proposal =
         serde_json::from_reader(file).expect("JSON was not well-formatted");
 
+    let client = HttpClient::new(args.tx.ledger_address.clone()).unwrap();
+
     let signer = WalletAddress::new(proposal.clone().author.to_string());
+    let goverance_parameters = rpc::get_governance_parameters(&client).await;
+    let current_epoch = rpc::query_epoch(args::Query {
+        ledger_address: args.tx.ledger_address.clone(),
+    })
+    .await;
+
+    if proposal.voting_start_epoch <= current_epoch
+        || proposal.voting_start_epoch.0
+            % goverance_parameters.min_proposal_period
+            != 0
+    {
+        println!("{}", proposal.voting_start_epoch <= current_epoch);
+        println!(
+            "{}",
+            proposal.voting_start_epoch.0
+                % goverance_parameters.min_proposal_period
+                == 0
+        );
+        eprintln!(
+            "Invalid proposal start epoch: {} must be greater than current \
+             epoch {} and a multiple of {}",
+            proposal.voting_start_epoch,
+            current_epoch,
+            goverance_parameters.min_proposal_period
+        );
+        safe_exit(1)
```

Comment:
let's skip the `safe_exit` call `if args.tx.force` is set (same at the other calls below)